src/herder/TransactionQueue.h

```diff
@@ -43,45 +43,54 @@ class TransactionQueue
     {
         SequenceNumber mMaxSeq{0};
         int64_t mTotalFees{0};
+        int mAge{0};
 
         friend bool operator==(AccountTxQueueInfo const& x,
                                AccountTxQueueInfo const& y);
     };
 
-    struct TxMap
+    struct AccountTransactions
     {
-        AccountTxQueueInfo mCurrentQInfo;
-        std::unordered_map<Hash, TransactionFramePtr> mTransactions;
-        void addTx(TransactionFramePtr);
-        void recalculate();
+        using Transactions = std::vector<TransactionFramePtr>;
+
+        int64_t mTotalFees{0};
+        int mAge{0};
+        Transactions mTransactions;
     };
-    typedef std::unordered_map<AccountID, std::shared_ptr<TxMap>> AccountTxMap;
 
     explicit TransactionQueue(Application& app, int pendingDepth, int banDepth);
 
     AddResult tryAdd(TransactionFramePtr tx);
-    // it is the responsibility of the caller to always remove such sets of
-    // transactions that remaining ones have their sequence numbers increasing
-    // by one
     void remove(std::vector<TransactionFramePtr> const& txs);
-    // remove oldest transactions and move all other transactions to slots older
-    // by one, this results in newest queue slot being empty
+
+    // remove transactions for accounts that have their pending transactions in
+    // this queue for at least pendingDepth
     void shift();
 
     AccountTxQueueInfo
     getAccountTransactionQueueInfo(AccountID const& accountID) const;
 
     int countBanned(int index) const;
     bool isBanned(Hash const& hash) const;
+
     std::shared_ptr<TxSetFrame> toTxSet(Hash const& lclHash) const;
 
   private:
+    using PendingTransactions =
```

Comment:
You need to put some comments on what is going on here: I had to stare for a while and reverse engineer what you're trying to do.

Reply:
I will add comments when all other issues are resolved, so I will not have to update it depending on chosen solutions.